src/main.rs

```diff
@@ -313,7 +313,7 @@ async fn main() -> Result<()> {
         if is_reboot || is_power_off || is_reload_config {
             loop {
                 {
-                    let retain_flag = retain_flag.lock().unwrap();
+                    let retain_flag = retain_flag.lock().expect("Mutex lock is safe because it only guards a simple boolean flag without complex operations.");
```

Comment:
I don't think this message sufficiently explains why we can always expect safe results.

Reply:
Initially, I used the “always safe” message in expect because the Mutex only protects a simple boolean flag, which makes it unlikely to panic within the lock scope.
However, after considering the possibility of other parts of the system panicking while holding the lock, I decided to replace expect with unwrap_or_else.
In case of poisoning, the value is now safely recovered, and a warning message is logged to aid debugging.